crates/router/src/connector/nmi/transformers.rs

```diff
@@ -1042,3 +1051,113 @@ pub struct SyncTransactionResponse {
 struct SyncResponse {
     transaction: SyncTransactionResponse,
 }
+
+#[derive(Debug, Deserialize)]
+pub struct RefundSyncBody {
+    order_id: String,
+    condition: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct NmiRefundSyncResponse {
+    transaction: RefundSyncBody,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NmiWebhookObjectReference {
+    pub event_body: NmiReferenceBody,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NmiReferenceBody {
+    pub order_id: String,
+    pub action: NmiActionBody,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NmiActionBody {
+    pub action_type: NmiActionType,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum NmiActionType {
+    Auth,
+    Capture,
+    Credit,
+    Refund,
+    Sale,
+    Void,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NmiWebhookEventBody {
+    pub event_type: NmiWebhookEventType,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum NmiWebhookEventType {
+    #[serde(rename = "transaction.sale.success")]
+    PaymentIntentSuccess,
```

Comment:
Instead of PaymentIntent in the enum naming, let's use SaleSuccess, SaleFailure, SaleUnknown